### PR TITLE
Support PHP8

### PR DESCRIPTION
**Pull request by brainexe**
there no BC visible to me :)

### CODE REVIEW

**Comment by Jeroeny** [comment]
Anything holding back this merge? :)

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ php:
   - 7.2
   - 7.3
   - 7.4
+  - nightly
```
Comment (Chi-teck) [inline .travis.yml:9]: As PHP 8 has just been released this probably can be changed to 8.0

Comment (brainexe) [reply .travis.yml:10]: yep, true. just added "8.0" 👍

**Comment by ktomk** [comment]
On Travis-CI there is yet no full support for `php: 8.0` - which is why this PR build still appears currently red to me - , PHP 8 support could work by `nightly`/`8.0snapshot` and/or using the standard Linux OS/dist (Xenial).

So for this PR if I may suggest: add an `allow_failures` job for `php: 8.0` (the recently added one; and keep everything else).

~~~
jobs:
  fast_finish: true
  allow_failures:
    - php: '8.0' # php 8.0 missing on travis <https://travis-ci.community/t/php-8-0-missing/10132>
~~~

And then test again without specifying `trusty` - if green, remove the allowed failure. See as well the link in the example above.

**Comment by ktomk** [comment]
@brainexe  Thanks for  taken care, I cherry picked the applicable changes in a different PR and closed this one. Thanks as well @Jeroeny.

**Pull request closed without merging**